src/backend/catalog/index.c

```diff
@@ -747,11 +748,14 @@ index_create(Relation heapRelation,
 	if (sql_dialect == SQL_DIALECT_TSQL && heapRelation->rd_rel->relpersistence == RELPERSISTENCE_TEMP)
 	{
 		/*
- 		 * Index of Table Variables should also be in ENR and the index
-		 * can only be created as part of DECLARE @[name] TABLE;
-		 * when specifying PRIMARY KEYs and/or CONSTRAINTs.
+		 * Index of temp objects in ENR should also be in ENR.
+		 * For table variables, index can only be created as
+		 * part of DECLARE @[name] TABLE; when specifying
+		 * PRIMARY KEYs and/or CONSTRAINTs.
 		 */
-		is_enr = (indexRelationName && strlen(indexRelationName) > 0 && indexRelationName[0] == '@');
+		is_enr = (indexRelationName && strlen(indexRelationName) > 0 &&
+			(indexRelationName[0] == '@' || (strchr(indexRelationName, '#') != NULL
+			&& get_ENR_withoid(currentQueryEnv, heapRelationId, ENR_TSQL_TEMP))));
```

Comment:
nitpick for performance purposes we do not need to call `get_ENR_withoid()` if it is a table variable.

Reply:
I don't think it's being called if it's a TV, there's parens around strchr+get_ENR_withoid

Reply:
okay got it thanks thats a bit painful to read :)

Reply:
Are we also allowed to create index for temp tables where the index name does not start with '#'?

Reply:
The strchr call is unnecessary, yeah